src/prefect/agent.py

```diff
@@ -193,19 +193,42 @@ async def submit_run(self, flow_run: FlowRun) -> None:
         if ready_to_submit:
             infrastructure = await self.get_infrastructure(flow_run)
 
-            try:
-                # Wait for submission to be completed. Note that the submission function
-                # may continue to run in the background after this exits.
-                await self.task_group.start(submit_flow_run, flow_run, infrastructure)
-                self.logger.info(f"Completed submission of flow run '{flow_run.id}'")
-            except Exception as exc:
-                self.logger.error(
-                    f"Flow runner failed to submit flow run '{flow_run.id}'",
-                    exc_info=True,
+            # Wait for submission to be completed. Note that the submission function
+            # may continue to run in the background after this exits.
+            await self.task_group.start(
+                self._submit_run_and_capture_errors, flow_run, infrastructure
+            )
+            self.logger.info(f"Completed submission of flow run '{flow_run.id}'")
+
+        self.submitting_flow_run_ids.remove(flow_run.id)
+
+    async def _submit_run_and_capture_errors(
+        self,
+        flow_run: FlowRun,
+        infrastructure: Infrastructure,
+        task_status: TaskStatus = None,
```

Comment:
It looks like `task_status`, will always be `None`. I'd drop it from the function definition and the call to `submit_flow_run` and then we can add it back when/if needed.

Reply:
`task_group.start` passes a `task_status`. It's a special keyword argument that lets the task group block until `.started()` is called.